apollo-runtime-kotlin/src/commonMain/kotlin/com/apollographql/apollo/interceptor/ApolloRequestInterceptor.kt

```diff
@@ -19,19 +21,22 @@ interface ApolloRequestInterceptor {
 }
 
 @ApolloExperimental
-internal class RealInterceptorChain private constructor(
+internal class RealInterceptorChain(
     private val interceptors: List<ApolloRequestInterceptor>,
-    private val index: Int
+    private val index: Int,
+    override val customScalarAdapters: CustomScalarAdapters,
```

Comment:
is there a reason we move customScalarAdapters from request to interceptor chain? I'm not 100% sure if this is a right place, before we can easily access them from request if request is not right place should we store them in execution context then?

Reply:
The idea was to make request a user-controlled type. The user would configure behaviour of the request there and create a call for it, much like OkHttp. Since `customScalarAdapters` are client-bound and not request-bound, this didn't feel like the right place. Right now it's reachable from the interceptor chain. Maybe ExecutionContext is better but I _think_ that would be less typed.